sklearn/metrics/scorer.py

```diff
@@ -67,6 +91,62 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """ Updates kwargs with ``self._pass_classes``
+
+        Checks if estimator has a classes argument and if scorer has a
+        ``self._pass_classes`` argument. If both exist, it return new kwargs
+        with ``self._pass_classes`` set as the classes_ attrbiute of estimator.
+        This is done only if ``self._pass_classes`` was not already set in
+        ``make_scorer`` initialization
+
+        Parameters
+        ----------
+        estimator : sklearn style estimator
+            The estimator from which classes_ attribute is to be extracted
+
+        target_type : string
+            The type of target y. Should be obtained using ``type_of_target``
+            from ``sklearn.utils.multiclass``
+        """
+
+        if self._pass_classes is None:
+            return self._kwargs
+
+        classes = getattr(estimator, "classes_", None)
+
+        # if classes_ parameter doesn't exist
+        if classes is None:
+            return self._kwargs
+
+        if self._pass_classes in signature(self._score_func).parameters:
```

Comment:
Hmm... I'm not sure I like this implicit magic. I'd rather have pass_classes=False (better than None) by default, and explicitly change our `make_scorer` invocations.

On the other hand, perhaps we are protecting users by making `pass_classes='labels'` by default as it makes `make_scorer` easier to use. Hmmm... I have to think.

Reply:
There is another advantage to have this as a string and not boolean. This will allow users who have different semantics for 'labels' in our API to use this functionality. eg, they might have 'present_classes' as an argument which works same as 'labels'. They can just pass 'present_classes' here and scorer will pass classes to that argument in metric call.